tests/playwright-test/playwright-test-fixtures.ts

```diff
@@ -146,7 +148,7 @@ async function runPlaywrightTest(childProcess: CommonFixtures['childProcess'], b
     const cwd = options.cwd ? path.resolve(baseDir, options.cwd) : baseDir;
     const packageRoot = path.resolve(baseDir, findPackageJSONDir(files, options.cwd ?? ''));
     const relativeBlobReportPath = path.relative(cwd, path.join(packageRoot, 'blob-report'));
-    const mergeResult = await mergeReports(relativeBlobReportPath, env, { cwd, additionalArgs });
+    const mergeResult = await mergeReports(relativeBlobReportPath, useIntermediateMergeReport ? envWithJsonReporter : env, { cwd, additionalArgs });
```

Comment:
```suggestion
    const mergeResult = await mergeReports(relativeBlobReportPath, envWithJsonReporter, { cwd, additionalArgs });
```

This line is already under if.